driver-core/src/main/com/mongodb/client/model/search/TextSearchOperator.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client.model.search;
+
+import com.mongodb.annotations.Evolving;
+
+/**
+ * @see SearchOperator#text(Iterable, Iterable)
+ * @since 4.7
+ */
+@Evolving
+public interface TextSearchOperator extends SearchOperator {
+    /**
+     * Creates a new {@link TextSearchOperator} that uses fuzzy search
+     * and does not use {@linkplain #synonyms(String) synonyms}.
+     *
+     * @param options Fuzzy search options.
+     * @return A new {@link TextSearchOperator}.
+     */
+    TextSearchOperator fuzzy(FuzzySearchOptions options);
```

Comment:
What would the implementation approach be if in the future we need to add a method to another interface that extends SearchOperator that has the same name and parameters as one in another interface that extends SearchOperator?  SearchConstructibleBsonElement couldn't implement both methods.

I guess we would just split SearchConstructibleBsonElement into multiple classes?

Reply:
Exactly. In fact, this has already happened with `FuzzySearchOptions`: methods `FuzzySearchOptions.option` and `SearchOptions.option` return `FuzzySearchOptions` and `SearchOptions` respectively. So I had to implement `FuzzySearchOptions` in `FuzzySearchOptionsConstructibleBson` instead of implementing it in `SearchConstructibleBson`.